include/device.h

```diff
@@ -302,6 +302,18 @@ void z_sys_device_do_config_level(s32_t level);
  */
 __syscall struct device *device_get_binding(const char *name);
 
+/**
+ * @brief device initialization success or not
+ *
+ * @param device Pointer to device structure.
+ *
+ * @return true if device successfully initialized; false if not.
+ */
+static inline bool is_device_initialized(struct device *device)
```

Comment:
Should be `device_is_initialized()`, we don't to introduce a new prefix that can conflict with other API.